### ISSUE TITLE
new hydrological parameters in Code Table 4.2

### ISSUE THREAD

**Issue by sebvi** [wmo-im/GRIB2#86]
# Branch
[not created yet]

# Summary and purpose
ECMWF is requesting new hydrological parameters for the EFAS, GLOFAS and ULYSSES projects.

# Action proposed
The team is kindly asked to review and approve the contents for inclusion within the next update to the WMO Manual on Codes.

# Discussions
ECMWF is operating several services for the Copernicus program in the domain of flood forecasting (EFAS/GLOFAS) and hydrological seasonal predictions (ULYSSES).
For these projects new (or correct) parameters are needed:

- snow melt accumulated over a time period: Currently, in code table 4.2, there is a parameter called "snow melt"  (discipline 0, category 1, entry 16) and a parameter called "upstream accumulated snow melt" (discipline 1, category 0, entry 15) both with units kg m-2 . These parameters should be deprecated because they already represents an accumulation (since the units are not kg m-2 s-1) and in principle can't be use with templates encoding statistical processing using code Table 4.10 (which changes units multiplying by time). We propose to deprecate both parameters and create a new parameter "snow melt rate" with units kg m-2 s-1 in discipline 1 category 0.
- accumulated runoff and drainage: Again, in code table 4.2, there are several "runoff", "runoff and drainage", etc. that are already accumulated parameters with units kg m-2. We would like to propose new parameters with units kg m-2 s-1 and deprecate the problematic parameters.
- Discharge return period with units m3 s-1: This is a statistical measure of levels of discharge that are expected with a certain frequency, for a example the 100-year return period for the Thames river is 187 m3/s of discharge.
- Upstream area, units m2 : The area which contributes with water to that specific point/grid box. This is a static variable.
- Drainage direction: The direction of flow from a point in the grid - This is an integer that describes the direction of flow from each point. The integer goes from 1-9 for each of the principal directions of a 3*3 grid starting from the mid point. a 5 denotes no direction. This is also a static variable

# Detailed proposal

| Discipline | Category | Code | Meaning | Units |
|--|--|--|--|--|
| 2 | 0 | 41 | Snow melt rate | kg m-2 s-1 |
| 2 | 0 | 42 | Water runoff and drainage rate | kg m-2 s-1 |
| 2 | 0 | 43 | Discharge return period | m3 s-1 |
| 2 | 0 | 44 | Drainage direction | integer |

**Comment by SimonElliottEUM** [comment]
![keypad](https://user-images.githubusercontent.com/62427578/115398066-0cd90a80-a1e7-11eb-8849-bfa31e50d0ab.jpeg)

**Comment by SimonElliottEUM** [comment]
As you can see, my key pad at home is the other way around

**Comment by lemkhenter** [comment]
Drainage direction: this parameter is provided by a hydraulic terrain model. This type of model is interested in the neighboring grid points to which the water drop can orient itself. It is therefore a 3 * 3 grid around the point in question.
So I agree with proposal.

**Comment by SimonElliottEUM** [comment]
Noting @lemkhenter 's comment, it looks like a code table defining the "direction" of the neighbouring grid will be best.  Not degrees.

**Comment by sebvi** [comment]
updated the proposal to take into acocund yesterday's comments.

**Comment by lemkhenter** [comment]
In my opinion, this proposal requires further study for several reasons:
1- the direction parameter, like the wind direction, requires the designation of an origin and a direction of evolution.
2- In the field of hydrology, the terrain models do not always have rectangular / square meshes. some models have a triangular mesh for example.
For these reasons and others on which specialists in the field can clarify, I propose to submit the proposal to a hydrology group for more detail and precision.

**Comment by sebvi** [comment]
Thank you @lemkhenter for your comments.

1- I think all directions we have (wind, waves, ocean currents, etc.) all have an origin at the centre of the grid box. Do you have an example where it is not the case? In the case of cardinal directions and not degrees, it seems to me that the "North" will be the North regardless of the origin chosen and the direction of evolution is implicitly "towards North" (because away from North would be then towards South). Would it help if we rename the parameter "Drainage Cardinal Direction"?
2- If a model has different requirements, for instance triangle mesh as you suggests with only 4 directions (3 sides of the triangle + no direction), there is nothing that prevents adding other entries in the new table. For instance one could imagine to "south south east", "east south east", "south south west", "west south west", etc.

**Comment by sebvi** [comment]
Note that if we rename the parameter Drainage cardinal direction, we restrict future use cases, for instance someone using "up" "Down", "left", "right" (although I would not be in favor of this sort o  use case!!)

**Comment by SimonElliottEUM** [comment]
@lemkhenter suggests to "submit the proposal to a hydrology group for more detail and precision". This seems like a wise move to me. We are an expert team on data representation - not hydrology. In a similar way we take satellite related topics to the CGMS Task Group on Satellite Data and Codes

**Comment by sebvi** [comment]
yes sure we can definitely do that. I don't have a contact point for an hydrology group (other than our own in-house group) so I am happy to take any suggestion.

**Comment by amilan17** [comment]
per meeting discussion: Upstream area will be added, A code table will be proposed for Drainage direction,

**Comment by jitsukoh** [comment]
@amilan17 @efucile I should have asked you earlier, but we need to consult this proposal to a relevant group of hydrological experts, which I don't know. Can you suggest which group we should consult and how? I understand there is a SC member from CHy: Mr Juan Bianchi, and there is also ET-HYDMON under INFCOM...

**Comment by sebvi** [comment]
I am happy to park for now parameters "Upstream area", "Drainage direction" and "Return period" until we receive a feedback from Hydrological teams. But I would really like to go forward with "Snowmelt rate" and "Water runoff and Drainage rate".

**Comment by amilan17** [comment]
per discussion at meeting: split move terms that need more consultation to new issue and update original comment with values that will go through FT2021-2. Also, provide link to new issue.

**Comment by sebvi** [comment]
removed the withdrawn parameters. I will move them to a new issue. Is there a way to move the valuable comments to the new issue?

**Comment by sebvi** [comment]
branch updated

**Comment by jitsukoh** [comment]
@sebvi @amilan17 I confirm that this proposal is finalized and the branch is updated, and move this issue to "Validated" status.

**Comment by amilan17** [comment]
@sebvi - Suggestion for notes, based on the proposal to deprecate terms. 

*   1-0, 15: Upstream accumulated snow melt (see Note 4)
    *   Note 4: It is recommended to use Snow melt rate instead (discipline 2, category 0, number 41).
*   0-1, 16: Snow melt (see Note 7) 
    *   Note 7: It is recommended to use Snow melt rate instead (discipline 2, category 0, number 41).

**Comment by sebvi** [comment]
@amilan17  I am happy with your suggestions.
